### DISCUSSION TITLE
MBP M1 Pro and dual LG Ultrafine 5K: "Use hardware DDC control" randomly enables itself.

### DISCUSSION THREAD

**Discussion by julioromano** [MonitorControl/MonitorControl#1384] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

Every now and them, completely at random, after waking the system from sleep or after disconnecting and reconnecting a display's thunderbolt cable I see that the "Use hardware DDC control" option on one of the 2 external displays is set to enabled even though I've never enabled it (it can be randomly either one of them when waking from sleep, otherwise if connecting and reconnecting one it happens on the one that has just been reconnected).
When this happens the MonitorControl menu shows the volume slider for that monitor and the brightness adjustment becomes "different" because when set to 0 it completely blacks out the display (which normally doesn't happen because this display's 0 brightness setting keeps the backlight at a very low level).

To fix this I have to go to the "Displays" tab and click "Reset settings" on the affected display. "Use hardware DDC control" will then go back to off, the volume slider disappears from the menu and the brightness adjustment goes back to normal.

It happens randomly on average 3 times per day during a full working day.

### Steps to reproduce

No steps to reproduce, unfortunately. I don't do anything particular when this bug happens.

### Expected behavior

"Use hardware DDC control" setting should not change by itself.

### Environment Information (please complete the following information)

```markdown
- macOS version: Macos 13.3.1 (a)
- Mac model: 2021 MacbookPro 16" M1 Pro 32GB 1TB
- MonitorControl version: 4.1.0 build 7034
- Monitor(s): dual LG Ultrafine 5k (first revision, without brightness sensor)
- Apple Silicon/M1 (yes or no): yes
```

**Comment by waydabber** [comment, marked as answer]
This might happen if the display's identifiers are changed. You can try [BetterDisplay](https://betterdisplay.pro) instead of MC and see if you have this issue. If so, then check the UUID value under the app's Settings/Displays/Show display information and see if it is different than usual or not.

<img width="942" alt="Screenshot 2023-05-31 at 16 41 30" src="https://github.com/MonitorControl/MonitorControl/assets/37590873/d4f8ebba-6ce9-4b1f-ba4b-7545f87010d5">

**Reply by julioromano** [reply]
I've upgraded to an M2 Pro MBP and (since it does not support DDC on M2) I don't have this weird behavior anymore, but I have another perhaps side effect that I hope might help you to trace the source of the glitch.

This is what happens randomly (same randomness as reported in the first message): A display is reported as `Control method: Software (gamma)`. Depending on the randomness sometime when waking from sleep or disconnecting/reconnecting it could also be both displays showing this.

<img width="702" alt="Screenshot 2023-06-06 at 09 44 11" src="https://github.com/MonitorControl/MonitorControl/assets/1273124/172fc5b9-bfd6-49c2-9586-c824d9559c86">

In the meantime I've installed BetterDisplay to check if the UUID changes. But it doesn't seem BetterDisplay allows for brightness syncing among displays, does it? Can I keep it active alongside MonitorControl or are they mutually exclusive?

**Reply by julioromano** [reply]
@waydabber got it, here's the display info when it's working correctly:
<img width="942" alt="ok" src="https://github.com/MonitorControl/MonitorControl/assets/1273124/d41f9618-895c-463b-aaf7-8efddc6612ef">

Here's the same info after a display disconnect/reconnect (when this happened MonitorControl was displaying the `Control method: Software (gamma)` warning as in the comment above): 
<img width="942" alt="error" src="https://github.com/MonitorControl/MonitorControl/assets/1273124/f70afd6f-43a8-451c-bba5-14af283013f8">

So apparently the UUID has not changed but the `configuration ID` did change and also the monitor is being reported as "Third-party external display" and not as "Apple external display". The capabilities section also changed.

Is it possible to debug further why this happens?

**Reply by julioromano** [reply]
After some further testing I realized that whenever this glitch happens, further disconnecting and reconnecting the affected display will not solve the issue unless I quit the MonitorControl app (and BetterDisplay was still running).
So it seems that for some reason MonitorControl remembers this wrong state and restores it whenever the display is reconnected.
To fix the issue I need to:
- Quit MonitorControl
- Disconnect/Reconnect the affected display
- Reopen MonitorControl

**Reply by waydabber** [reply]
It seems somehow the mac and display can't negotiate properly and the display is not recognized properly as a display that supports Apple's brightness own control protocol. I am not sure why this happens though. :(

**Reply by julioromano** [reply]
It happens only when MonitorControl is running. If I quit it and keep only BetterDisplay it doesn't seem to happen.

**Reply by julioromano** [reply]
Oh I was wrong, today it happened again, with only BetterDisplay running. So this issue seems unrelated with either BetterDisplay or MonitorControl :(
